js/modules/k6/http/request.go

```diff
@@ -84,6 +84,24 @@ func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*Re
 		}
 		return &Response{Response: r, client: c}, nil
 	}
+	if callback != nil {
+		callbackF := c.moduleInstance.vu.RegisterCallback()
+		go func() {
+			resp, err := httpext.MakeRequest(c.moduleInstance.vu.Context(), state, req) //nolint:govet
+			if err != nil {
+				callbackF(func() error {
+					_, e := callback(goja.Undefined(), nil, c.moduleInstance.vu.Runtime().ToValue(err))
+					return e
+				})
+			}
+			c.processResponse(resp, req.ResponseType)
+			callbackF(func() error {
+				_, e := callback(goja.Undefined(), c.moduleInstance.vu.Runtime().ToValue(resp))
+				return e
+			})
+		}()
+		return nil, nil //nolint:nilnil
+	}
```

Comment:
This is a new branch so we should cover it with some basic unit tests

Reply:
This code definitely won't end in the final code. It's here only so that the `http_callback.js` can be used for illustration purposes.

Reply:
WDYM by "final code"? You mean in the upcoming release?

I don't think we should release a feature publicly only to remove it soon after.

And does this affect `responseCallback` in any way?

Reply:
the commit containing this change will not be merged in master - it's here only to illustrate the use in case you want to try locally.

`callback` and `responseCallback` do two completely different things and completely different times. 
- `responseCallback` - changes `expected_response` value. This is done before the response is returned or metrics are emitted
- `callback` - will be called after the request is done with the `response`/`error` that was produced. 

And again as the `callback` not being merged at all it won't affect anything

I am going to move this commit to a separate branch and link to it.

Reply:
this has now been moved to https://github.com/grafana/k6/commit/61aa0e5103df7e139d4df932b0457f42ece4eef3